app/controllers/Checkout.scala

```diff
@@ -11,17 +11,30 @@ import play.api.libs.json._
 import play.api.mvc._
 import services.AuthenticationService.authenticatedUserFor
 import services.CheckoutService.CheckoutResult
-import services.TouchpointBackend.Resolution
 import services._
 import utils.TestUsers.{NameEnteredInForm, PreSigninTestCookie}
 import views.html.{checkout => view}
 
+import scala.Function.const
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 object Checkout extends Controller with LazyLogging {
+  object SessionKeys {
+    val subsName = "newSubs_subscriptionName"
+    val ratePlanId = "newSubs_ratePlanId"
+    val userId = "newSubs_userId"
+    val token = "newSubs_token"
```

Comment:
These field names should be capitalised:

    "Constant names should be in upper camel case. That is, if the member is final, immutable
    and it belongs to a package object or an object, it may be considered a constant (similar to
    Java’s static final members):"

http://docs.scala-lang.org/style/naming-conventions.html#values_variable_and_methods



Also, token should be more specifically named: ie 'IdentityGuestPasswordSettingToken'